src/documentation/sdks/euphoria.md

```diff
@@ -18,72 +18,530 @@ See the License for the specific language governing permissions and
+```java
+BeamFlow flow = ...
+Dataset<String> dataset = ...
+
+Dataset<String> mapped =
+  MapElements
+    .named("MapThem")
+    .of(dataset)
+    .using(
+      (input, context) -> {
+        // use simple counter
+        context.getCounter("my-counter").increment();
+
+        return input.toLowerCase();
+        })
+      .output();
+```
+Accumulators are translated into Beam Metrics in background so they can be viewed the same way. Namespace of translated metrics is set to operator's name.
+
+### Windowing
+Euphoria follows the same [windowing principles]({{ site.baseurl }}/documentation/programming-guide/#windowing) as Beam Java SDK. Every shuffle operator, that is every operator which needs to shuffle elements over network between executors, allows you to set it. The same parameters as in Beam are required. `WindowFn`, `Trigger` and `WindowingStrategy`. Users are guided to either set all three or none when building an operator. Windowing is propagated down through `BeamFlow` the same way as it works with `Pipeline`.
+```java
+CountByKey.of(input)
+  .keyBy(e -> e)
+  .windowBy(FixedWindows.of(Duration.standardSeconds(1)))
+  .triggeredBy(DefaultTrigger.of())
+  .discardingFiredPanes()
+  .output();
+```
+
+### Integration of Euphoria into existing pipelines
+`EuphoriaPTransform` allows to define composite `PTransform` so Euphoria can be seamlessly integrated to already existing Beam `Pipelines`. User only need to provide implementation of function which takes input `Dataset`  and outputs another `Datatset`. The input dataset is nothing else than mirror of a input `PCollection`. Output `Dataset` is transformed to `Pcollection` automatically.
+```java
+//suppose inputs PCollection contains: [ "a", "b", "c", "A", "a", "C", "x"]
+PCollection<KV<String, Long>> lettersWithCounts =
+  inputs.apply("count-uppercase-letters-in-Euphoria",
+    EuphoriaPTransform.of(
+      (Dataset<String> input) -> {
+        Dataset<String> upperCase =
+          MapElements.of(input)
+            .using((UnaryFunction<String, String>) String::toUpperCase)
+            .output();
+
+        return CountByKey.of(upperCase).keyBy(e -> e).output();
+    }));
+//now the 'lettersWithCounts' will conntain [ KV("A", 3L), KV("B", 1L), KV("C", 2L), KV("X", 1L) ]
+```
+
+## How to get Euphoria
+Euphoria is located in `dsl-euphoria` branch of Apache Beam project. To build `euphoria` subproject call:
+```
+./gradlew beam-sdks-java-extensions-euphoria-core:build
+```
+
+## Operator Reference
+Operators are higher level transformations of data which allows to build whole logic of data processing jobs. All the Euphoria operators are documented in this section including examples. There are no examples with [windowing](#windowing) applied for the sake of simplicity. Refer to the [windowing section](#windowing) for more details.
```

Comment:
Operators are basically higher level data transformations, which allows you to build business logic of your data processing job in a simple way.

All of the Eu...

Reply:
Accepted